builtin/game/item.lua

```diff
@@ -458,7 +476,8 @@ function core.do_item_eat(hp_change, replace_with_item, itemstack, user, pointed
 				itemstack:add_item(replace_with_item)
 			else
 				local inv = user:get_inventory()
-				if inv:room_for_item("main", {name=replace_with_item}) then
+				-- Check if inv is null, since non-players don't have one
+				if inv and inv:room_for_item("main", {name=replace_with_item}) then
```

Comment:
Why is the `inv` check required here? The functions around require a valid player anyway.

Reply:
I don't get what you mean, everything used up to this point would work with non-player ObjectRefs, wouldn't it?

Reply:
Oh whoops, I forgot to check if it's nil.

Reply:
Wait no I don't because core.item_eat already checks for nil. But it could still be a non-player ObjectRef.